### ISSUE TITLE
feat(deploy): support Angular Universal

### ISSUE THREAD

**Issue by damienwebdev** [zeit/ng-deploy-now#9]
It looks like out-of-the-box this doesn't support Angular Universal. I have a few apps that utilize Server-Side-Rendering via. Universal with Now.

The app code is open-source, https://github.com/graycoreio/daffodil/tree/develop/apps/daffio/serverless and all I end up doing is publishing this via package via the `now` cli. If there was built-in support for `ng deploy` that'd be amazing.

**Comment by rsns** [comment]
Great! Is there a way how to deploy clean Angular Universal installation to zeit?

**Comment by VigneshVaidyanathan** [comment]
I am trying to deploy Angular Universal on Zeit/Vercel as well. But havent found a proper way to do it yet. Were you able to deploy it @rsns Please let me know if you have done it

**Comment by rsns** [comment]
I choosed Scully.io, it's simple, provides all ssr benefits, and Zeit has
guide on how to deploy with it.

**Comment by VigneshVaidyanathan** [comment]
@rsns Nice. Unfortunately, I need to get data from API during the runtime based on the Subomain like user.domain.com and render the page correspondingly. Guess thats not easy to be prerendered and use Scully in Zeit.

**Comment by rsns** [comment]
@VigneshVaidyanathan I understand.
I have a Scully and Angular app that makes a call to an API, also from a subdomain, API call is fired in app.components OnInit lifecycle hook. After getting data from API - app is rendering components on page based on that data, and no errors, but data from API will not be server side rendered, I guess.

> In a static site, we don’t make any API call to get the data from the server, rather we put all the content on our pages as data and text and make it available to our end users, like blog posts or portfolios.

This is a quote from [Scully, the First Static Site Generator for Angular](https://indepth.dev/scully-the-first-static-site-generator-for-angular/)

**Comment by VigneshVaidyanathan** [comment]
@rsns Exactly. That is my problem. I figured a way to get the API calls done on the server and passing it to the component when rendering the view. So the API call is made in the server and rendered in the browser, but I am facing some issues in deployment.

**Comment by tawakol** [comment]
Anything ever happened with this? It's super important and others do provide the ability (netlify, DO, etc)

**Comment by cyrus-za** [comment]
[This](https://vercel.com/guides/deploying-scully-with-vercel) might be of use for anyone stumbling on this issue

**Comment by damienwebdev** [comment]
Just chiming in here that this is in-fact possible. It takes a bit of configuration to do. Unfortunately, I don't have the time to write down an entire tutorial but this [vercel.json](https://github.com/graycoreio/daffodil/blob/develop/apps/daffio/serverless/vercel/vercel.json) file works. You do need to make a change in the UI of Vercel in order for it to work as expected though.


<img width="796" alt="Screen Shot 2021-08-11 at 10 50 33 AM" src="https://user-images.githubusercontent.com/9029654/129070322-fd5d199a-fa8b-46cd-9666-6c3af41a2108.png">

**Comment by jdgamble555** [comment]
I tried something like this, although I think I am close, to no avail.  Anyone have any ideas?

```typescript
{
  "version": 2,
  "public": true,
  "name": "test-universal",
  "builds": [
    { "src": "package.json", "use": "@vercel/static-build" },
    { "src": "./dist/test/server/main.js", "use": "@vercel/node" }
  ],
  "routes": [{ "src": "/(.*)", "dest": "./dist/test/server/main.js" }]
}
```

J

**Comment by damienwebdev** [comment]
@jdgamble555 I have several projects using Universal that are currently deployed to Vercel:

* https://next.graycore.io
* https://next.daff.io
* https://daff.io

If you peruse through the OS codebase I linked above, you will find everything you need to deploy Universal to Vercel.

**Comment by jdgamble555** [comment]
Thanks @damienwebdev.  I've been working on this for weeks, and it definitely helped fill in the blanks.  Here is the solution:

https://dev.to/jdgamble555/how-to-deploy-angular-universal-to-vercel-31d0
